include/onnxruntime/core/graph/graph.h

```diff
@@ -968,6 +1029,33 @@ class Graph {
 
   void ToGraphProtoInternal(ONNX_NAMESPACE::GraphProto& graph_proto) const;
 
+  template <typename TInstance>
+  static auto GetProducerNodeImpl(
+      TInstance& instance, const std::string& node_arg_name)
+      -> decltype(instance.GetNode(0)) {
+    auto iter = instance.node_arg_to_producer_node_.find(node_arg_name);
```

Comment:
nit: this could probably be on one line and be a bit more readable. same for GetConsumerNodeImpl. just need to keep lines under 120